packages/algolia-experiences/src/setup-instantsearch.ts

```diff
@@ -75,6 +63,17 @@ function registerComponents(search: InstantSearch, settings: Settings) {
     disconnectedCallback() {
       search.removeWidgets(this.widgets);
     }
+
+    attributeChangedCallback(name: string, oldValue: string, newValue: string) {
+      if (oldValue === null || name !== 'experience-id') {
+        return;
+      }
+
+      if (oldValue !== newValue) {
+        this.disconnectedCallback();
+        this.connectedCallback();
```

Comment:
for later: the order should be fetch -> remove -> add, not remove -> fetch -> add, but I see how this current code is much simpler

Reply:
Could be simple, I could wrap this into `fetchConfiguration(newValue).then` and it would pick up the cached configuration in `connectedCallback`

Reply:
also good for me :)